src/OptionSettlement.sol

```diff
@@ -488,6 +428,121 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
         return optionRecord;
     }
 
+    /// @dev Performs fair exercise assignment by pseudorandomly selecting a claim
+    /// bucket between the intial creation of the option type and "today". The buckets
+    /// are then iterated from oldest to newest (looping if we reach "today") if the
+    /// exercise amount overflows into another bucket. The seed for the pseudorandom
+    /// index is updated accordingly on the option type.
+    function _assignExercise(uint160 optionId, Option storage optionRecord, uint112 amount) internal {
+        // A bucket of the overall amounts written and exercised for all claims
+        // on a given day
+        ClaimBucket[] storage claimBucketArray = _claimBucketByOption[optionId];
```

Comment:
If a claim bucket is depleted, we should pop the last element off the array and overwrite no? Otherwise we can keep hitting empty buckets. The other issue with using an array is that we end up with out of order data.

Reply:
buckets are only added if we've crossed the day boundary, and options can only be added to the last/most recent bucket, so the data will be ordered the same way. 

ordering only becomes a problem if we swap/pop. we could consider having some sort of mapping from option to a list of unexercised buckets that we maintain during the exercise process. an index is selected randomly from that array for exercise.